subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/ClassTestResults.java

```diff
@@ -58,7 +64,14 @@ public String getName() {
         return name;
     }
 
+    public String getDisplayName() {
+        return displayName;
+    }
+
     public String getSimpleName() {
+        if (displayName != null && !displayName.equals(name)) {
+            return displayName;
```

Comment:
❌ Why do we return the display name here? This method is clearly meant to return the simple name of the class.

Reply:
Because display name may be the same as the "name", which is full qualified name and is actually not simple.

Reply:
What I'm saying is that the word "simple name" has a clear meaning - it's the last segment of the class name. It should not return the display name. Both examples above should return "Test" as the simple name.

Reply:
It would be interesting to know what we even use this for. Maybe the method name was badly chosen.

Reply:
Okay, that makes sense. I'll create a new method `getReportName()` to handle this case.

Reply:
This method is used to render index failed tests, for example:

```
[Failed tests tab]

FailedTestClass1.failTestMethod()
FailedTestClass2.failTestMethod()
```

And the class result page breadcrumb:

```
All > org.gradle > TestClass
```

Reply:
It would be great if at the end of this PR you could show some screenshots of the html report after this change, given some JUnit 5 tests with custom names.

Reply:
Done.